### PR TITLE
[Face] Add snapshot sample and WPF sample under newly created app-samples folder.

### PR DESCRIPTION
**Pull request by lebronJ**
## Purpose
1. Create an `app-samples` folder for samples not fitting into the existing `samples` folder.
2. Add snapshot sample under `app-samples/Face/FaceApiSnapshotSample`.
3. Add WPF sample under `app-samples/Face/Cognitive-Services-Face-WPF`.
4. Bump up Face SDK version to latest 2.3.0 preview for all face samples.

## Other Information
Adjust `.gitignore` to skip nuget packages.

### CODE REVIEW

**Comment by lebronJ** [comment]
@yangyuan Could you help check this PR? We'd like to refer to this Snapshot sample from Face API' s how-to-guide documentation page.

File: app-samples/Face/Cognitive-Services-Face-WPF/Sample-WPF/packages.config [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Microsoft.Azure.CognitiveServices.Vision.Face" version="2.3.0-preview" targetFramework="net452" />
```
Comment (yangyuan) [inline app-samples/Face/Cognitive-Services-Face-WPF/Sample-WPF/packages.config]: also change this if you change Framework

File: app-samples/Face/FaceApiSnapshotSample/FaceApiSnapshotSample/Program.cs [hunk 2]
```diff
@@ -0,0 +1,266 @@
+﻿namespace FaceApiSnapshotSample
```
Comment (yangyuan) [inline app-samples/Face/FaceApiSnapshotSample/FaceApiSnapshotSample/Program.cs]: This one seems to be compatible with auto-test framework, any reason to be a standalone application sample?

Comment (lebronJ) [reply app-samples/Face/FaceApiSnapshotSample/FaceApiSnapshotSample/Program.cs]: For snapshot feature, there is another required user provided id `Azure Cognitive Service Face subscription id`. The existing samples for auto-test needs only `Azure Cognitive Service Face subscription key`.

File: app-samples/Face/Cognitive-Services-Face-WPF/Sample-WPF/App.config [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+    <startup> 
+        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
```
[the remark this answers is not in the document]
Comment (lebronJ) [reply app-samples/Face/Cognitive-Services-Face-WPF/Sample-WPF/App.config]: Have not realized this, just directly migrate our deprecated sample without changing this. If you think this is recommended, I am fine with this change.

[the remark this answers is not in the document]
Comment (lebronJ) [reply app-samples/Face/Cognitive-Services-Face-WPF/Sample-WPF/App.config]: Upgraded to `4.6`.

File: app-samples/Face/Cognitive-Services-Face-WPF/Sample-WPF/packages.config (hunk 1 above)
Comment (lebronJ) [reply app-samples/Face/Cognitive-Services-Face-WPF/Sample-WPF/packages.config]: Upgraded to `4.6`.

File: app-samples/Face/FaceApiSnapshotSample/FaceApiSnapshotSample/Program.cs (hunk 2 above)
Comment (yangyuan) [reply app-samples/Face/FaceApiSnapshotSample/FaceApiSnapshotSample/Program.cs]: okay, I believe this should not be a limitation. But I need some tests.
I'll let the code in first.

Comment (lebronJ) [reply app-samples/Face/FaceApiSnapshotSample/FaceApiSnapshotSample/Program.cs]: Thanks. As I said, one more parameter should be passed into the snapshot sample if you adapt it to existing auto-test framework.

**Pull request merged**